### ISSUE TITLE
Properly print 'None' for option types from FSI

### ISSUE THREAD

**Issue by baronfel** [fsharp/fslang-suggestions#364]
# Properly print 'None' for option types from FSI [6690744]

**Submitted by TeaDrivenDev _ on 11/9/2014 12:00:00 AM**  
**18 votes on UserVoice prior to migration**  

FSI output properly handles 'Some x', but often prints 'None' as 'null', apparently when the value is part of a larger type.

```fsharp
let x : uint16 option = None
```

correctly results in: 

```fsharp
val x : uint16 option = None
```

while

```fsharp
let y : string * uint16 option = "", None
```

instead prints: 

```fsharp
val y : string * uint16 option = ("", null)
```
## Response

*\* by fslang-admin on 2/5/2016 12:00:00 AM **

This should certainly be fixed, as much as possible
Don Syme, F# Language Evolution

**[Original UserVoice Submission](https://fslang.uservoice.com/forums/245727-f-language/suggestions/6690744)**
[Archived Uservoice Comments](../tree/master/archive/suggestion-6690744-properly-print-none-for-option-types-from-fsi.md#comments)

**Comment by ForNeVeR** [comment]
There's also an issue with `None.GetType()`:
```fsharp
> None.GetType();;
System.NullReferenceException: Object reference not set to an instance of an object.
   at <StartupCode$FSI_0002>.$FSI_0002.main@()
Stopped due to error
```

**Comment by AviAvni** [comment]
Fixed

**Comment by dsyme** [comment]
Thanks! @AviAvni   Closing as this has been implemented in F# 4.2
